actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -231,6 +231,9 @@ class Parameters
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
     def initialize(parameters = {})
+      unless parameters.respond_to?(:with_indifferent_access)
```

Comment:
Ah I see, yes that might get confusing 🤔 are you not worried about the `respond_to?` implementation in terms of performance?